js/src/common/extend.ts

```diff
@@ -1,3 +1,10 @@
+export type KeyOfType<T extends object, Type> = Exclude<
```

Comment:
This is a bit hard to follow. Am I correct in understanding that:

1. We take all keys of T (each locally K)
2. We map each K => K (if T[K] = T)
3. We take those values with `[keyof T]`
4. We filter out everything that doesn't with Exclude

Could we make this a bit easier to digest by assembling this type operator step by step? The variable names for each intermediate step would then serve as documentation.